Volume/Views/MainView/MagazineReaderView.swift

```diff
@@ -0,0 +1,222 @@
+                        //                            userData.isMagazineSaved(magazine) ?
+                        //                            VolumeEvent.bookmarkMagazine.toEvent(.magazine, value: magazine.id, navigationSource: navigationSource) :
+                        //                                VolumeEvent.unbookmarkMagazine.toEvent(.magazine, value: magazine.id, navigationSource: navigationSource)
+                        //                        )
+                    }, label: {
+                        Image(systemName: userData.isMagazineSaved(magazine) ? "bookmark.fill" : "bookmark")
+                            .font(Font.system(size: 18, weight: .semibold))
+                            .foregroundColor(.volume.orange)
+                    })
+                    //                    .disabled(bookmarkRequestInProgress)
+                    
+                    Spacer()
+                        .frame(width: 16)
+                    
+                    Button {
+                        //                    TODO: Uncomment after implementing shareMagazine for analytics
+                        //                        AppDevAnalytics.log(VolumeEvent.shareMagazine.toEvent(.magazine, value: magazine.id, navigationSource: navigationSource))
+                    } label: {
+                        Image(systemName: "square.and.arrow.up.on.square")
+                            .font(Font.system(size: 16, weight: .semibold))
+                            .foregroundColor(.volume.orange)
+                    }
+                    
+                    Spacer()
+                        .frame(width: 16)
+                    
+                    Button {
+                        incrementShoutouts(for: magazine)
+                    } label: {
+                        Image.volume.shoutout
+                            .resizable()
+                            .scaledToFit()
+                            .frame(height: 24)
+                            .foregroundColor(.volume.orange)
+                            .foregroundColor(isShoutoutsButtonEnabled ? .volume.orange : .gray)
+                    }
+                    .disabled(!isShoutoutsButtonEnabled)
+                    
+                    Spacer()
+                        .frame(width: 6)
+                    
+                    Text(String(max(magazine.shoutouts, userData.shoutoutsCache[magazine.id, default: 0])))
+                        .font(.helveticaRegular(size: 12))
+                }
+            }
+            .padding([.leading, .trailing], 16)
+            .padding([.top, .bottom], 8)
+            .background(Color.white)
+        }
+        
+        var body: some View {
+            ZStack {
+                VStack(spacing: 0) {
+                    Spacer()
+                        .frame(height: showToolbars ? 40 : 20)
+                    if let magazineUrl = magazine.pdfUrl {
```

Comment:
Instead of making it `MagazineReaderView`'s responsibility to handle optionals, you may want to consider blocking the user from opening this view unless `pdfURL` is non-nil. What's the point of opening a magazineReader that has no magazine, anyway? This change should minimize the logic in this view a bit.